tests/foreman/api/test_activationkey.py

```diff
@@ -10,7 +9,24 @@
 from robottelo.test import APITestCase
 
 
-@ddt
+def _good_max_content_hosts():
+    """Return a generator yielding valid ``max_content_hosts`` values."""
+    return (gen_integer(*limits) for limits in ((1, 20), (10000, 20000)))
```

Comment:
The benefit to generators is that you're defining an algorithm for producing values, not the values themselves. This is costly:

```python
l = [i for i in range(1000000)]
```

But this is cheap:

```python
g = (i for i in range(1000000))
```

This allows for more "streamy", less "chunky" code.